python/lsst/afw/table/_syntax.py

```diff
@@ -202,3 +202,70 @@ def processArray(a):
         else:
             d[name] = processArray(self.get(schemaItem.key))
     return d
+
+def BaseCatalog_asAstropy(self, cls=None, copy=False, unviewable="copy"):
+    """!
+    Return an astropy.table.Table (or subclass thereof) view into this catalog.
+
+    @param[in]   cls        Table subclass to use; None implies astropy.table.Table itself.
+                            Use astropy.table.QTable to get Quantity columns.
+
+    @param[in]  copy        Whether to copy data from the LSST catalog to the astropy table.
+                            Not copying is usually faster, but can keep memory from being
+                            freed if columns are later removed from the Astropy view.
+
+    @param[in]  unviewable  One of the following options, indicating how to handle field types
```

Comment:
On the philosophical side, when I see `Table(lsst_table)` I almost view as a casting operation and it seems obvious to me that this should work.  So one just needs to provide the machinery to cast from one kind of table to other. 

Right now `Table` has methods `to_pandas()` and `from_pandas()` to convert back and forth to pandas DataFrames.  I think this is really inelegant compared to `DataFrame(astropy_table)` and `Table(dataframe)`.  Part of the reason is that `Table(dataframe)` can actually take all the other constructor args, for instance I could change the column names or types at that time.  With `from_pandas()` currently you don't have the full constructor API.  It is certainly possible to put the full API into `from_pandas()` but that just gets ugly and repetitious for each new compatible table type.